sources/api/early-boot-config/src/main.rs

```diff
@@ -306,11 +307,59 @@ impl PlatformDataProvider for AwsDataProvider {
     }
 }
 
-/// This function determines which provider we're currently running on.
+#[cfg(bottlerocket_platform = "aws-dev")]
+struct LocalFileDataProvider;
```

Comment:
I haven't looked through the rest yet, but why restrict the file provider to aws-dev?  It's there for testing in odd situations, I'd say, and odd situations can arise in any variant.  Is there a reason not to allow it in other variants?